students/Wayne/Assignments/Lesson3/Mailroom.py

```diff
@@ -1,51 +1,84 @@
 """
-Simple mailroom program for a non profit 
+Simple mailroom program for a non profit
 """
 
-#Start with interactive loop 
+# Start with interactive loop
 
 import sys
+from textwrap import dedent
 
-donors = [("Donor A",[50,100,24,100]),
-         ("Donor B",[525,10,23]),
-         ("Donor C",[50]),
-        ]
+donors = [("Donor A", [50, 100, 24, 100]),
+          ("Donor B", [525, 10, 23]),
+          ("Donor C", [50]),
+          ]
+
+"""
+#Building Donor List
+"""
+
+
+def list_donors():
+    print("Donors:\n")
+    for donor in donors:
+        print(donor[0])
 
 
 def thankyou():
-    print("This is the thank you\n")
+    while True:
+        name = input("Enter the donor's name or a list of donor's"
+                     "names to see all donors (or type 'menu'"
+                     "to exit)>").strip()
+        if name == 'list':
+            list_donors()
+        elif name == 'menu':
+            return
+        else:
+            break
+
+# creates loop for donation amount input
+
+    while True:
+        donationinput = input("Enter the amount donated "
+                              "(or 'menu' to exit)>").strip()
+        if donationinput == 'menu':
+            return
+
+# Building reporting function
+
 
 def report():
-    print ("This is the report function\n")
+    print("This is the report function\n")
+
+# Builds the quit function to exit the application
+
 
 def quit():
     sys.exit()
 
-def mainloop():
-    print("Welcome to Mailroom")
 
-    while True:
-        print("What do you want to do?")
-        print("1)Thank You\n"
-              "2)Report\n"
-              "3)Quit\n")
-        response = input('>>')
-
-        print("response was:", response)
-        if response not in ("1","2","3"):
-            print("Not a Valid Response, input 1, 2, or 3***\n")
-            continue
-        elif response == "1":
-            thank_you()
-        elif response == "2":
-            report()
-        elif response == "3":
-            quit()
-            
+def mainloop():
 
+    mainmenuinput = input(dedent("""
+        What would you like to do?
 
+        (1) - Send a Thank You Letter
+        (2) - Build a Report
+        (3) - Quit
 
-if __name__ == "__main__":
-    mainloop()
+        > """))
+    return mainmenuinput.strip()
 
 
+if __name__ == "__main__":
+    running = True
+    while running:
+        response = mainloop()
+        if response not in ('1', '2', '3'):
+            print("Not a Valid Response, input 1, 2, or 3")
+            continue
+        elif response == '1':
+            thankyou()
+        elif response == '2':
+            report()
+        elif response == '3':
+            quit()
```

Comment:
A solid start, review the solutions for a complete solution and make sure you understand the logic flow and how to get it all done